components/apimgt/org.wso2.carbon.apimgt.rest.api.common/src/main/resources/publisher-api.yaml

```diff
@@ -8881,6 +8881,191 @@ paths:
             "https://127.0.0.1:9443/api/am/publisher/v4/apis/01234567-0123-0123-0123-012345678901/detach-labels"'
       operationId: detachLabelsFromAPI
 
+  ######################################################
+  # API themes related APIs
+  ######################################################
+  /apis/{apiId}/api-themes:
+    get:
+      operationId: getApiThemes
+      summary: Retrieve API themes
+      description: Returns the list of API themes and their publish status for the given API.
+      parameters:
+        - name: publish
+          in: query
+          description: Filter themes based on published status
+          required: false
+          schema:
+            type: boolean
+        - name: apiId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: List of API themes
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
```

Comment:
We should map 404 as well

Reply:
Addressed by [8th set of required changes](https://github.com/wso2/carbon-apimgt/pull/12874/commits/3eefad006cdd2122381e6ff1706f1e9650d1b938)